src/neptune/new/internal/backends/hosted_neptune_backend.py

```diff
@@ -522,7 +526,7 @@ def _execute_upload_operations(
                     swagger_client=self.leaderboard_client,
                     container_id=container_id,
                     attribute=path_to_str(op.path),
-                    source=op.file_path,
+                    source=op.get_absolute_path(operation_storage),
```

Comment:
Looks like both - `clear` and `get_absolute_path` should be defined in OperationStorage. Operation should be a simple serializable dataclass with minimum logic. And it would be better to pass operation object instead of raw path as an argument.

Reply:
But if we define `clear` and `get_absolute_path` in `OperationStorage` we'll have to pass `Operation` (in particular `UploadFile`) to those functions because we would like to clear or get path of particular operation.

At the same time we need `upload_path` of `OperationStorage` when creating operation (when we create `UploadFile` of `in-memory` file we also save file to `.neptune` catalog).

It'll require `OperationStorage` to be some kind of proxy on `UploadFile` operation irresponsible also for creating it.

If you're ok with that I can do that, what do you think?